### DISCUSSION TITLE
Only serialize java.io.Serializable classes

### DISCUSSION THREAD

**Discussion by nixoso** [eclipse-serializer/serializer#163] [Q&A] [answered]
Is there a way to serialize only classes that inherit from _java.io.Serializable_ and simply ignore the others, like a filter?  

The reason for this is that I would like to see what is possible with the _eclipse-serializer_ compared to the [standard serializer](https://github.com/apache/wicket/blob/master/wicket-core/src/main/java/org/apache/wicket/serialize/java/JavaSerializer.java) from the Apache Wicket Framework.

**Comment by fh-ms** [comment, marked as answer]
You can add a filter with the `PersistenceTypeEvaluator`

```java
SerializerFoundation<?> foundation = SerializerFoundation.New()
	.setTypeEvaluatorPersistable(Serializable.class::isAssignableFrom);
Serializer<byte[]> serializer = Serializer.Bytes(foundation);
```

**Comment by nixoso** [comment]
Thanks, it works up to the _proxy classes_.
`org.eclipse.serializer.persistence.exceptions.PersistenceException: Proxy classes (subclasses of java.lang.reflect.Proxy) are not supported.`
It could have worked 😉
